### PR TITLE
added consul query api support(also post method)

### PR DESCRIPTION
**Pull request by shalev67**

### CODE REVIEW

**Comment by abn** [comment]
@shalev67 great work; would it be possible to add a few sanity test cases? Standard agent should suffice.

**Comment by dropje86** [comment]
Waiting very anxious for this to be merged and released ;)

**Comment by shalev67** [comment]
I've added a basic sanity test and upgraded the binaries to 0.6.4

**Comment by dropje86** [comment]
Nice, thnx @shalev67! Hopefully this'll be merged soon..

**Comment by abn** [comment]
@shalev67 @dropje86 I have rebased, and added some extra changes for this one and prepped the merge at #112 - this is for a cleaner merge. Closing this one out in favor of the revised PR.

File: consul/base.py [hunk 1]
```diff
@@ -1838,3 +1839,210 @@ def peers(self):
...
+            token = token or self.agent.token
+            if token:
+                params['token'] = token
+            if dc:
+                params['dc'] = dc
+
+            return self.agent.http.get(
+                lambda x: json.loads(x.body), '/v1/query', params=params)
+
+        def create(self, service,
```
Comment (cablehead) [inline consul/base.py]: I'm pretty sure mashing the features of Consul's query into a flat function signature is going to come unstuck. For example, if more DNS options besides ttl are added, or if additional template types are added.

I think we'll want something similar to the Check conveniences to allow succinct expression of the full complexity of the query api. Not a huge issue immediately, except for backwards compatibility if we try and address this later. /cc @abn fyi..

**Pull request closed without merging**